src/robot/libraries/Collections.py

```diff
@@ -1277,60 +1317,74 @@ def _report_error(default_message, message, values=False):
 
 class Normalizer:
 
-    def __init__(self, ignore_case=False, ignore_order=False, ignore_keys=None):
-        self.ignore_case = ignore_case
+    def __init__(
+        self,
+        ignore_case: "IgnoreKeyCase | IgnoreValueCase | bool" = False,
+        ignore_order: bool = False,
+        ignore_keys: "Sequence[Hashable] | None" = None,
+    ):
         if isinstance(ignore_case, str):
             self.ignore_key_case = ignore_case.upper() not in ("VALUE", "VALUES")
             self.ignore_value_case = ignore_case.upper() not in ("KEY", "KEYS")
         else:
-            self.ignore_key_case = self.ignore_value_case = self.ignore_case
+            self.ignore_key_case = self.ignore_value_case = ignore_case
+        self.ignore_case = ignore_case
         self.ignore_order = ignore_order
         self.ignore_keys = self._parse_ignored_keys(ignore_keys)
 
-    def _parse_ignored_keys(self, ignore_keys):
+    def _parse_ignored_keys(
+        self,
+        ignore_keys: "Sequence[Hashable] | None",
+    ) -> "set[Hashable]":
         if not ignore_keys:
             return set()
-        try:
-            if isinstance(ignore_keys, str):
-                ignore_keys = literal_eval(ignore_keys)
-            if not is_list_like(ignore_keys):
-                raise ValueError
-        except Exception:
-            raise ValueError(
-                f"'ignore_keys' value '{ignore_keys}' cannot be converted to a list."
-            )
         return {self.normalize_key(k) for k in ignore_keys}
 
-    def normalize(self, value):
+    @overload
+    def normalize(self, value: str) -> str: ...
+
+    @overload
+    def normalize(self, value: Mapping) -> Mapping: ...
+
+    @overload
+    def normalize(self, value: Sequence) -> Sequence: ...
+
+    @overload
+    def normalize(self, value: ListLike) -> ListLike: ...
```

Comment:
This seems unnecessary. Mapping and Sequence convert 2/3 of ListLike, the final part would be Set.